scvi/nn/_base_components.py

```diff
@@ -293,8 +293,9 @@ def forward(self, x: torch.Tensor, *cat_list: int):
         q = self.encoder(x, *cat_list)
         q_m = self.mean_encoder(q)
         q_v = self.var_activation(self.var_encoder(q)) + self.var_eps
-        latent = self.z_transformation(reparameterize_gaussian(q_m, q_v))
-        return q_m, q_v, latent
+        dist = Normal(q_m, q_v.sqrt())
+        latent = self.z_transformation(dist.rsample())
+        return dist, latent
```

Comment:
this is a breaking change, some people might rely on this code. I would add a `return_dist` option, set it False and then make it true in each module class

Reply:
gotcha